base/markdown/render/plain.jl

```diff
@@ -47,20 +86,26 @@ function plaininline(io::IO, md...)
     end
 end
 
-plaininline(io::IO, md::Vector) = plaininline(io, md...)
 
-plaininline(io::IO, md::Image) = print(io, "![$(md.alt)]($(md.url))")
+plaininline(md::Vector) = join(map(plaininline, md))
+
+plaininline(md::Image) = "![$(plaininline(md.alt))]($(md.url))"
 
-plaininline(io::IO, s::String) = print(io, s)
+plaininline(md::Link) = "[$(plaininline(md.text))]\($(md.url)\)"
 
-plaininline(io::IO, md::Bold) = plaininline(io, "**", md.text, "**")
+plaininline(s::String) = s
 
-plaininline(io::IO, md::Italic) = plaininline(io, "*", md.text, "*")
+plaininline(md::Bold) = "**" * plaininline(md.text) * "**"
 
-plaininline(io::IO, md::Code) = print(io, "`", md.code, "`")
+plaininline(md::Italic) = "*" * plaininline(md.text) + "*"
 
+plaininline(md::Code) = "`$(md.code)`"
+
+plaininline(io::IO, md::Vector) = plaininline(io, md...)
+plaininline(io::IO, md::Union(Image, String, Bold, Italic, Code, Link, Image)) = print(plaininline(md))
```

Comment:
Constructing the string is going to be slow here, so it's best to do it the other way around – leave the `plaininline(io...` methods as they are and have a `plaininline(x) = sprint(plaininline, x)` method if it's not there already.